var/spack/repos/builtin/packages/plasma/package.py

```diff
@@ -98,24 +125,20 @@ def edit(self, spec, prefix):
         # make sure CC variable comes from build environment
         make_inc.filter("CC *[?]*= * .*cc", "")
 
-        libs = self.getblaslapacklibs().ld_flags
-        if 'readline' in self.spec:
-            libs += ' ' + self.spec['readline'].libs.ld_flags
-            libs += ' ' + find_system_libraries(['libdl']).ld_flags
-        make_inc.filter("LIBS *[?]*= * .*", "LIBS = " + libs)
-
     @property
     def build_targets(self):
         targets = list()
 
+        if self.spec.satisfies('@18.9.0:'):
```

Comment:
@luszczek It looks like this is intended to only return build targets for `17.1`. However, I don't see anything making use of `build_targets` for `17.1`, i.e. it looks like the entire build is defined in the `install` method. I'm wanted to check that (a) in fact you want these constraints to apply to `17.1` and (b) I was curious if you saw them getting applied (e.g. in the build log).

From looking at this, I think the return value of this method is never used, so you could remove all of the logic here.

Reply:
I don't know if I'm misusing `build_targets` but `install` method is trivial. It is `edit` that does the real work as much as it can. What is left is things I cannot edit through code and I can just stick on the invocation line of `make`. And this is exactly what I'm trying to do in `build_targets` for 17.1. More specifically, `MKLROOT` is not nicely available for automated editing so I override it in `build_targets` and `make MKLROOT=-lmkl` does the rest.

Reply:
@scheibelp Did I justified `build_targets` logic? Could you please approve the PR?

Reply:
I tested `17.1` with supported dependencies and they work with the property `build_targets` removed. So I removed it with another commit. Thank you for checking.